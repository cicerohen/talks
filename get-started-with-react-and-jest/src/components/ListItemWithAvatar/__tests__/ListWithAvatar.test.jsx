import React from 'react';
import { shallow } from 'enzyme';
import ListItemWithAvatar from '../ListItemWithAvatar';

const minimalProps = {
  children: <div />,
  avatarUrl: 'https://avatars1.githubusercontent.com/u/819041?s=400&u=e1678422ed2e3d88571273fcfeedc8647ea0c282&v=4',
};

describe('<ListItemWithAvatar /> Tests', () => {
  it('should render without exploding', (done) => {
    const wrapper = shallow(
      <ListItemWithAvatar {...minimalProps}/>
    );
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly', (done) => {
    const wrapper = shallow(
      <ListItemWithAvatar {...minimalProps}/>
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
