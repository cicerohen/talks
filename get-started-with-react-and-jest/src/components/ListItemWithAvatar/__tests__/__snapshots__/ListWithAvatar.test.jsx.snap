// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemWithAvatar /> Tests should render correctly 1`] = `
<ListItem
  style={
    Object {
      "container": "container",
      "root": "root",
    }
  }
>
  <Avatar
    src="https://avatars1.githubusercontent.com/u/819041?s=400&u=e1678422ed2e3d88571273fcfeedc8647ea0c282&v=4"
    style={
      Object {
        "root": "avatar__root",
      }
    }
  />
  <div
    className="body"
  >
    <div />
  </div>
</ListItem>
`;
